File: src/main/java/net/matees/settings/global/PVPEnabled.java
package net.matees.settings.global;

import org.bukkit.Material;

import net.matees.settings.BooleanSetting;
import net.matees.settings.Global;

public class PVPEnabled extends BooleanSetting implements Global {

    private static final PVPEnabled INSTANCE = new PVPEnabled();
    private Boolean setting = true;

    private PVPEnabled() {
    }

    public static PVPEnabled getInstance() {
        return INSTANCE;
    }

    @Override
    public void onChange() {
    }

    @Override
    public String getName() {
        return "PVP Enabled";
    }

    @Override
    public String getDescription() {
        return "Enable or disable PVP";
    }

    @Override
    public Boolean getSetting() {
        return this.setting;
    }

    @Override
    public void setSetting(Boolean setting) {
        this.setting = setting;
    }

    @Override
    public Material getMenuItemMaterial() {
        return Material.IRON_SWORD;
    }

    @Override
    public int getMenuItemSlot() {
        return 14;
    }

}

File: src/main/java/net/matees/settings/global/PVPEnabled.kt
package net.matees.settings.global

import net.matees.settings.AbstractBooleanSetting
import net.matees.settings.IGlobal
import org.bukkit.Bukkit
import org.bukkit.Material

class PVPEnabled private constructor() : AbstractBooleanSetting(), IGlobal {
    override var setting: Boolean? = true

    override fun onChange() {
        val world = Bukkit.getWorld("world")
        if (world != null) {
            world.pvp = this.setting!!;
        }
    }

    override val name: String
        get() = "PVP Enabled"

    override val description: String
        get() = "Enable or disable PVP"

    override val menuItemMaterial: Material
        get() = Material.IRON_SWORD

    override val menuItemSlot: Int
        get() = 14

    companion object {
        val instance: PVPEnabled = PVPEnabled()
    }
}
